lib/config/flat-options.js

```diff
@@ -182,7 +183,6 @@ const flatOptions = npm => npm.flatOptions || Object.freeze({
   strictSSL: npm.config.get('strict-ssl'),
   defaultTag: npm.config.get('tag'),
   get tag () {
-    log.warn('FIXME', 'using tag option, should be defaultTag')
```

Comment:
nit: I thought we wanted this? At least until we have more stability in npm@7?

Reply:
The problem is that it warns every time we do `...npm.flatOptions`, which we do a lot of times.  And we can't make the option non-enumerable, or else it won't be carried over in those cases.  We'd have to do a custom clone function that does `Object.getOwnPropertyDescriptors` and turns the setter into a setter on the resulting object, and then use that everywhere, but then our deps _also_ do `...opts` all over the place, so actually making that change would be almost as bad as figgy-pudding.

It was a reasonable idea, but doesn't actually work, turns out.

Reply:
Totally makes sense, I hadn't realised that. Completely agree we should just pull this out for now. Might be worth just commenting it, so we remember where our warnings were?

Reply:
Sure, if you wanna, go for it.  I was just gonna leave it as a pre-v7 todo item to remove the getters from here and grep for any usage of those field names.